h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java

```diff
@@ -112,19 +114,25 @@ public void cleanUp() {
   public List<String> calcTypeaheadMatches(String filter, int limit) {
     final String input = GcsBlob.removePrefix(filter);
     final String[] bk = input.split("/", 2);
-    List<String> results = new ArrayList<>();
+    List<String> results = limit > 0 ? new ArrayList<String>(limit) : new ArrayList<String>();
```

Comment:
Setting it to limit could allocate too many resources in the backing array in vain in some cases. Not sure if we want it. However, it makes sense to do it. On the other hand, allocating an empty ArrayList with zero-length backing array will probably result in immediate array enlargment in vain I believe. This component is not performance-critical, but I'd think set the initial capacity to 10 for example, or some arbitrary number > 2.

Reply:
When you use the default non-parametric constructor it doesn't allocate anything (it reuses a static 
"empty" array). The initialization of the backing array will actually happen after you add the first element to the array.

I think this code is fine as it is